tests/Form/GeneralTest.php

```diff
@@ -435,4 +354,36 @@ public function testSubmitEmptyTextarea()
             $this->assertStringContainsString($searchString, $pageContent);
         }
     }
+
+    /**
+     * @dataProvider provideInvalidValues
+     * @param mixed $value
+     */
+    public function testSetInvalidValueInField(string $field, $value): void
+    {
+        $this->getSession()->visit($this->pathTo('/advanced_form.html'));
+
+        $webAssert = $this->getAssertSession();
+
+        $color = $webAssert->elementExists('named', ['id_or_name', $field]);
+
+        $this->expectException(DriverException::class);
+        $color->setValue($value);
+    }
+
+    public static function provideInvalidValues(): Generator
+    {
+        $scenarios = [
+            'about' => [null, true, false, ['bad']],
+            'notes' => [null, true, false, ['bad']],
+            'first_name' => [null, true, false, ['bad']],
+            'submit' => [null, true, false, ['bad'], 'update'],
+        ];
+
+        foreach ($scenarios as $field => $values) {
+            foreach ($values as $value) {
+                yield "$field field with " . var_export($value, true) => [$field, $value];
```

Comment:
the generated dataset names are bad here:

- the description of the field should be about the type of input, not about the id used in the fixtures
- the case of an array value should talk about an array value, not about `array(0 => 'bad')` (or even worse by having newlines in it).

I'd like to keep good names for named datasets for 2 reasons:

- they are a better fit than the values for a skipping logic in driver configs
- they are part of the report for PHPUnit failures, so they contribute to understand what the failing test is about (especially when the method name becomes more generic)

Reply:
agreed on all counts